slsim/Sources/Supernovae/supernovae_lightcone.py

```diff
@@ -0,0 +1,72 @@
+from skypy.galaxies.redshift import redshifts_from_comoving_density
+from slsim.Sources.Supernovae.supernovae_pop import SNIaRate
+from astropy.table import Table
+from astropy import units
+
+
+class SNeLightcone(object):
+    """Class to integrate SNe comoving density n(z) in light cone volume."""
+
+    def __init__(self, cosmo, redshifts, sky_area, noise, time_interval):
+        """
+        :param cosmo: cosmology object
+        :type cosmo: ~astropy.cosmology object
+        :param redshifts: redshifts for supernovae density lightcone to be evaluated at
+        :type redshifts: array-like
+        :param sky_area: sky area for sampled galaxies in [solid angle]
+        :type sky_area: `~Astropy.units.Quantity`
+        :param noise: poisson-sample the number of galaxies in supernovae density lightcone
+        :type noise: bool
+        :param time_interval: time interval for supernovae density lightcone to be evaluated over
+        :type time_interval: `~Astropy.units.Quantity`
+        """
+        self._cosmo = cosmo
+        self._redshifts = redshifts
+        self._sky_area = sky_area
+        self._noise = noise
+        self._time_interval = time_interval
+
+        sne_rate = SNIaRate(self._cosmo, self._redshifts[-1])
+        self.density = self.convert_density(
+            sne_rate.calculate_SNIa_rate(self._redshifts)
+        )
+
+    def convert_density(self, density):
```

Comment:
somewhere the little h got missed, or where do you take it into account?

Reply:
For some reason I keep forgetting to put it in the units and then forgot to divide by the factor. I divided by the factor in the initialization function when `self.density` is calculated.